store/postgres/src/relational_queries.rs

```diff
@@ -494,30 +500,34 @@ impl EntityData {
         match self.data {
             j::Object(map) => {
                 let typname = std::iter::once(self.entity).filter_map(move |e| {
-                    if remove_typename {
-                        None
-                    } else {
+                    if T::WITH_INTERNAL_KEYS {
                         Some(Ok((Word::from("__typename"), T::Value::from_string(e))))
+                    } else {
+                        None
                     }
                 });
                 let entries = map.into_iter().filter_map(move |(key, json)| {
                     // Simply ignore keys that do not have an underlying table
                     // column; those will be things like the block_range that
                     // is used internally for versioning
                     if key == "g$parent_id" {
```

Comment:
I know this is not from this PR but these names should defo be placed in constants, including the lazy Word const IMO